dotCMS/src/main/java/com/dotcms/security/apps/AppsAPIImpl.java

```diff
@@ -753,38 +767,89 @@ private boolean validateServiceDescriptor(final AppSchema appDescriptor)
+                    "Param `%s`: is missing required field `type` (STRING|BOOL|SELECT) .",
                     name));
         }
 
         if (!UtilMethods.isSet(descriptor.getRequired())) {
-            errors.add(String.format("Param `%s`: is missing required field `required` (true|false) .",
-                    name));
+            errors.add(
+                    String.format("Param `%s`: is missing required field `required` (true|false) .",
+                            name));
         }
 
         if (!UtilMethods.isSet(descriptor.getHidden())) {
             errors.add(
-                    String.format("Param `%s`: is missing required field `hidden` (true|false) .", name));
+                    String.format("Param `%s`: is missing required field `hidden` (true|false) .",
+                            name));
         }
 
-        if (Type.BOOL.equals(descriptor.getType()) && UtilMethods.isSet(descriptor.getHidden()) && descriptor.isHidden()) {
+        if (UtilMethods.isSet(descriptor.getValue()) && StringPool.NULL
+                .equalsIgnoreCase(descriptor.getValue().toString()) && descriptor.isRequired()) {
             errors.add(String.format(
-                    "Param `%s`: Bool params can not be marked hidden. The combination (Bool + Hidden) isn't allowed.",
-                    name));
+                    "Null isn't allowed as the default value on required params see `%s`. ",
+                    name)
+            );
         }
 
-        if (Type.BOOL.equals(descriptor.getType()) && UtilMethods.isSet(descriptor.getValue())
-                && !isBoolString(descriptor.getValue())) {
-            errors.add(String.format(
-                    "Boolean Param `%s` has a default value `%s` that can not be parsed to bool (true|false).",
-                    name, descriptor.getValue()));
+        if (Type.BOOL.equals(descriptor.getType())) {
+            if (UtilMethods.isSet(descriptor.getHidden())
+                    && descriptor.isHidden()) {
+                errors.add(String.format(
+                        "Param `%s`: Bool params can not be marked hidden. The combination (Bool + Hidden) isn't allowed.",
+                        name));
+            }
+
+            if (UtilMethods.isSet(descriptor.getValue())
+                    && !isBoolString(descriptor.getValue().toString())) {
+                errors.add(String.format(
+                        "Boolean Param `%s` has a default value `%s` that can not be parsed to bool (true|false).",
+                        name, descriptor.getValue()));
+            }
         }
 
-        if (StringPool.NULL.equalsIgnoreCase(descriptor.getValue()) && descriptor.isRequired()) {
-            errors.add(String.format(
-                    "Null isn't allowed as the default value on required params see `%s`. ",
-                    name)
-            );
+        if(Type.STRING.equals(descriptor.getType())){
+            if(!(descriptor.getValue() instanceof String)){
```

Comment:
Codacy found an issue: [These nested if statements could be combined](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5895122)